### PR TITLE
release: v9.61.2

### LINKED ISSUE #817: validate_model_allowed() has no allowlist arm for ollama/copilot/vibe — OCTOPUS_*_ALLOWED_MODELS silently ignored
`validate_model_allowed()` in `scripts/lib/dispatch.sh:779-799` has allowlist-var arms for codex, gemini, agy, claude-sdk, claude, openrouter, atlascloud, openai-compatible*, perplexity, qwen, cursor-agent, commandcode, and opencode, then falls through:

```bash
*)          return 0 ;;  # Unknown provider — allow
```

`ollama`, `copilot`, and `vibe` are not in that case, so any `OCTOPUS_OLLAMA_ALLOWED_MODELS` / `OCTOPUS_COPILOT_ALLOWED_MODELS` / `OCTOPUS_VIBE_ALLOWED_MODELS` a user sets is silently ignored — the model restriction never applies for these three providers.

Same class of gap as #799 ("unknown agents fail open"), but a different function/file (`dispatch.sh` allowlist validation vs. `model-resolver.sh` availability checks).

Surfaced by CodeRabbit review on #816 while fixing the *separate* `OCTOPUS_*_MODEL` pin-resolution gap for these same three providers. Confirmed this allowlist gap is **not** introduced or changed by #816 — before that fix, `provider` was empty for these agent types and also hit the same `*) return 0` fallback, so the allowlist-bypass behavior is identical before and after. Filing separately rather than scope-creeping #816, since this is model-restriction enforcement (a security/policy concern), not model-pin resolution.

Suggested fix: add the three missing case arms, following the existing pattern, e.g.:
```bash
ollama)     allowlist_var="OCTOPUS_OLLAMA_ALLOWED_MODELS" ;;
copilot)    allowlist_var="OCTOPUS_COPILOT_ALLOWED_MODELS" ;;
vibe)       allowlist_var="OCTOPUS_VIBE_ALLOWED_MODELS" ;;
```
Per `docs/PROVIDERS.md` item 3, this is one of several sites in the provider-wiring checklist — worth checking against that checklist rather than doing it in isolation.

### LINKED ISSUE #818: Codex 9.44 doctor can self-link the stable plugin path
## Summary

The Codex-distributed Claude Octopus skill at `~/.codex/plugins/cache/nyldn-plugins/claude-octopus/9.44.0/skills/skill-doctor/SKILL.md` still uses:

```bash
ln -sfn "$OCTO_PLUGIN_ROOT" "${HOME}/.claude-octopus/plugin"
```

When `OCTO_PLUGIN_ROOT` resolves to the already-valid stable link itself, this replaces the link with a self-reference. Subsequent Octopus commands fail with:

```text
bash: ~/.claude-octopus/plugin/scripts/orchestrate.sh: Too many levels of symbolic links
```

## Reproduction

1. Install/use the Codex Octopus bundle at v9.44.0.
2. Have `~/.claude-octopus/plugin` point to a valid current Claude plugin cache.
3. Invoke the `skill-doctor` resolver with no `CLAUDE_PLUGIN_ROOT` set.
4. The resolver selects the stable link as `OCTO_PLUGIN_ROOT`, then rewrites that link to point to itself.

## Current upstream state

Current Octopus 9.61.0 and `main` already contain the `_octo_stable` physical-path guard introduced in #782, so the source fix exists. The remaining problem is that the Codex-distributed 9.44.0 package still exposes the unsafe resolver.

## Requested fix

Please refresh or republish the Codex plugin distribution with the guarded resolver, or backport the guard to the supported Codex package. A regression test should cover the case where `OCTO_PLUGIN_ROOT` is the stable symlink itself.

## Local recovery verified

Pointing the stable link back to the 9.61.0 cache and backporting the guard locally restored `doctor hooks`: `hooks.json` valid, all 41 hook scripts valid. Plugin validation and the hook trap/compatibility suites also pass.

### LINKED ISSUE #822: Unit template test rewrites the active checkout during make ci-local
## Summary

`tests/unit/test-skill-templates.sh` runs `scripts/gen-skill-docs.sh` directly against `PROJECT_ROOT`. The generator overwrites tracked `.claude/skills/*/SKILL.md` files from stale templates, so a successful `make ci-local` leaves the checkout dirty and can silently replace current skill behavior with older generated content.

## Reproduction

From a clean checkout:

```bash
git status --short
make ci-local
git status --short
```

Observed after the otherwise-green gate:

```text
 M .claude/skills/flow-define/SKILL.md
 M .claude/skills/flow-deliver/SKILL.md
 M .claude/skills/flow-develop/SKILL.md
 M .claude/skills/flow-discover/SKILL.md
```

The test contains both the direct production-tree generation call and a stale-marker append:

```bash
"$GEN_SCRIPT" > /dev/null 2>&1
echo "# stale marker" >> "$(resolve_claude_skill_path "flow-discover")"
"$GEN_SCRIPT" > /dev/null 2>&1
```

This is especially risky because the same file comments say the flow skills are directly authored since v9.10.2. In an isolated CI worktree, the generator still resolved and rewrote the active task checkout, demonstrating that the test is not hermetic.

## Proposed fix

Run generator mutation checks against a temporary copied fixture/root, add a regression assertion that the tracked source hashes do not change, and retain read-only structural checks against the real checkout.

### LINKED ISSUE #825: fix: council approval gates ignore non-interactive session signals
## Problem

`council_prompt_gate_approval` prompts solely on `[[ -t 0 && -t 1 ]]`. It ignores the repository's established non-interactive signals (`CI`, `OCTOPUS_NON_INTERACTIVE`, remote/autonomous session flags).

As a result, PTY-backed automation and local `make ci-local` runs can stop at:

```text
Gate A: accept council synthesis? [y/N]
```

even when `OCTOPUS_NON_INTERACTIVE=1` is set.

## Reproduction

```bash
printf 'n\n' | script -q /tmp/octopus-council-prompt.log \
  bash -c 'source scripts/lib/council.sh; OCTOPUS_NON_INTERACTIVE=1 council_prompt_gate_approval gate-a "Gate A: accept council synthesis?"'
```

The captured output contains the prompt.

## Expected

Approval gates should remain closed without prompting when any established headless/remote/autonomous signal is active. Explicit `OCTOPUS_COUNCIL_APPROVED_GATES` should continue to work.

## Acceptance criteria

- `council_prompt_gate_approval` respects `CI`, `OCTOPUS_NON_INTERACTIVE`, Claude remote/web session flags, Octopus remote/autonomous flags, and TTY availability.
- PTY-backed tests prove non-interactive signals suppress reads/prompts.
- Interactive approval behavior and explicit approved-gate behavior remain covered.

### LINKED ISSUE #827: fix: lifecycle hook timeout fallback depends on missing pkill
## Problem

The lifecycle-hook timeout regression has two independent faults:

1. `tests/unit/test-agent-lifecycle-events.sh` measures promptness with wall-clock `date +%s`. During a long local suite the clock can jump, producing false elapsed values (observed 16s, then 208s and 999s while the tool returned much sooner).
2. The built-in timeout branch in `_octopus_agent_lifecycle_event` claims to work without external timeout utilities, but descendant teardown uses `pkill -P` without checking that `pkill` exists. On a minimal PATH the wrapper returns, but a delayed hook child survives and continues running.

## Evidence

The original full gate failure was:

```text
lifecycle hook uses built-in timeout fallback when timeout commands are unavailable
FAIL: built-in timeout fallback did not return promptly
```

Repeated isolated runs exposed the wall-clock instability. A stronger mutation test then made the hook spawn a delayed child marker with both `timeout` and `pkill` absent: old code returned but the child later wrote the marker, proving the process-tree leak independently of elapsed-time reporting.

## Expected

- Promptness assertions use a monotonic per-shell timer rather than wall-clock time.
- The built-in path terminates and reaps the observer process tree within bounded grace using dependencies it actually detects.
- No orphaned hook child survives teardown.

## Acceptance criteria

- No undeclared `pkill` dependency in the built-in lifecycle-hook fallback.
- Regression runs with `timeout`, `pgrep`, and `pkill` unavailable but portable `ps` available.
- Slow hook returns within configured timeout plus bounded teardown grace.
- Delayed descendant marker is never written after return.
- Hook failure remains non-fatal and the focused test emits no missing-function noise.

### LINKED ISSUE #829: fix: release validator rejects registered nested skills
## Problem

`scripts/validate-release.sh 9.61.2` reports all four registered `skills/octopus-starter-pack/*/SKILL.md` entries as missing even though the files exist and are registered in `.claude-plugin/plugin.json`.

The validator builds `SKILL_FILES` with `find skills -mindepth 2 -maxdepth 2 -name SKILL.md`, which sees ordinary `skills/<skill>/SKILL.md` layouts but cannot see the supported nested `skills/octopus-starter-pack/<skill>/SKILL.md` layout. The registered-skill comparison therefore fails a valid release with four false errors.

## Expected

The release validator should compare recursive skill paths relative to `skills/`, while still ignoring unrelated files and requiring an exact `SKILL.md` registration match. Add a regression test covering both ordinary and nested skill directories.

## Reproduction

```bash
./scripts/validate-release.sh 9.61.2
```

Observed on current `main` after #828; the check exits 1 with four `Registered skill ... does not exist` errors.

### LINKED ISSUE #835: fix(careful): scope destructive SQL detection to executable context
## Problem

`hooks/careful-check.sh` scans the entire Bash command string for destructive SQL phrases. Even after requiring `TRUNCATE` statement shape, read-only commands still produce confirmation prompts:

```text
grep truncate somefile        -> Destructive SQL prompt
rg "DROP TABLE" src          -> Destructive SQL prompt
printf "TRUNCATE users"       -> Destructive SQL prompt
```

This is a Claude Octopus hook false positive, not a host/harness error. It interrupts governed and unattended runs for harmless source searches and output commands.

## Durable acceptance criteria

- Read-only search/output contexts such as `grep truncate somefile`, `rg "DROP TABLE" src`, and `printf "TRUNCATE users"` stay quiet.
- Known SQL-client execution (`psql -c "TRUNCATE users"`, `mysql -e "DROP TABLE users"`) still asks.
- Direct destructive SQL statements still ask.
- Piped SQL into a known client still asks.
- Behavioral regression tests invoke the real hook with careful mode active.

PR #831 fixes related substring/boundary cases and is the intended implementation vehicle for this context-aware hardening.

### LINKED ISSUE #836: docs(update): surface Claude auto-update opt-in for third-party marketplace
## Problem

Claude Code supports startup auto-update for installed marketplace plugins, but official documentation says third-party marketplaces have auto-update disabled by default. The Octopus README currently gives manual update commands without telling stale-version users how to opt into host-managed automatic updates.

A plugin should not rewrite its loaded cache or user enablement state. The durable path is to use the host updater and keep the marketplace identity/version contract stable.

## Acceptance criteria

- Update/troubleshooting guidance states that `nyldn-plugins` auto-update is opt-in in Claude Code.
- Give the UI path: `/plugin` -> Marketplaces -> `nyldn-plugins` -> Enable auto-update.
- Retain the explicit `claude plugin marketplace update` and `claude plugin update` fallback commands.
- Explain that a reload/restart applies an updated plugin.
- Include the guidance in v9.61.2 release notes/update fix.

Source: https://code.claude.com/docs/en/discover-plugins#configure-auto-updates

### LINKED ISSUE #840: Provider smoke-test failures do not exclude failed providers from `probe` dispatch
## Summary

On v9.61.1, a provider can fail the run's smoke test and still be selected and spawned by `probe_discover`.

This is related to the currently open Gemini degradation report #815, but the issue here is the routing contract after a known smoke-test failure, not the fact that Gemini is degraded.

## Observed sequence

Provider detection reported Gemini degraded. The workflow then emitted:

```text
Running provider smoke test...
Gemini: Smoke test failed (unknown error)
Smoke test: degraded mode (2/3 providers passed)
probe_discover: smart dispatch=5:codex,gemini,agy,copilot,claude-sonnet:high
```

The event stream subsequently recorded both:

```text
provider.selected provider=gemini
agent.spawned agent_type=gemini task_id=probe-...-1
```

The Gemini result was only a header/empty-output stub in this interrupted run.

## Why this matters

A smoke test appears to validate that a provider can actually answer in the current environment. Continuing to route work to a provider that just failed that validation can waste latency or quota, reduce useful perspective coverage, and make the "degraded mode" warning misleading.

The impact is more visible now that Gemini degradation is an expected environment state on some hosts (#772 and #815).

## Code-path observation

`provider_smoke_test` returns success whenever at least one provider passes, even when others fail. Its per-provider failures are not propagated into an exclusion set consumed by `get_dispatch_strategy` / `probe_discover`, so the later dispatch still uses configured/available providers.

## Expected behavior

After smoke testing, one of these contracts should be explicit:

1. Recommended: exclude providers that failed the current smoke test from dispatch for that run, while continuing in degraded mode with providers that passed.
2. Alternatively: document smoke testing as informational only and avoid messaging that implies failed providers have been gated.

## Suggested acceptance test

Configure two fake providers where one passes smoke and one fails. Run a probe and assert that degraded mode continues successfully but only the passing provider appears in `provider.selected` / `agent.spawned` events.

### LINKED ISSUE #846: ci: macOS unit matrix exceeds 15-minute job timeout
## Failure

Release PR #823 run 31411222295 canceled the macOS unit job at the configured 15-minute ceiling. The log shows individual suites still passing and then `The operation was canceled` during `test-provider-registry-contracts.sh`; Ubuntu, smoke, and symlink jobs passed. This is infrastructure budget exhaustion, not a test assertion failure.

## Cause

`.github/workflows/test.yml` says the macOS suite completes in just over 10 minutes and sets `timeout-minutes: 15`. The current 248-suite matrix needed more than 15 minutes on this runner, so the job was killed before completing.

## Acceptance criteria

1. Give the unit matrix enough headroom for current slower macOS runners.
2. Add a regression contract so the timeout cannot silently return to the exhausted 15-minute budget.
3. Rerun the failed jobs and require the complete release matrix to pass.

Discovered while finalizing v9.61.2; fix should be included in that release.

### PR DESCRIPTION
**Pull request by nyldn**
## Release v9.61.2

This release rolls the hook reliability, stale-install update recovery, model routing, provider health, and release-validation fixes into one tested artifact.

### Hook reliability

- Careful-mode guards now recognize executable context and token boundaries, eliminating the recurring `rg`/`grep`/documentation, dotfile, subpath, and word-substring false positives while retaining destructive-command protection (#831, fixes #835).
- Lifecycle hook timeouts no longer depend on wall-clock `SECONDS`, GNU `timeout`, or `pkill`; the built-in watchdog owns and reaps the entire process group, with static and mutation-proven cleanup contracts (#828, #833, #845; closes #827, #832, #837, #844).
- Non-interactive council approval gates no longer prompt merely because automation inherited terminal signals (#826; closes #825).
- Skill-template tests generate in an isolated copy, removing a test-side source of hook and working-tree noise (#824; closes #822).

### Stale plugin updates

- Restores the stable `claude-octopus@nyldn-plugins` Codex marketplace identity and validates both Claude and Codex selectors during shared-marketplace sync (#820; closes #818).
- Documents the Claude host-managed auto-update opt-in for third-party marketplaces, manual update/reload recovery, and the Codex marketplace upgrade flow. Octopus does not rewrite its live cache or enablement state.

### Provider and release hardening

- Ollama, Copilot, and Vibe now honor model pins and allowlists (#816, #819; closes #817).
- Unknown agent IDs fail closed while supported Grok, Command Code, Atlas Cloud, and Vibe seats retain explicit auth contracts (#842; bounded part of #799).
- Smoke-failed providers are excluded from smart/full/minimal fleets, and successful recovery clears the lockout immediately (#843; closes #840).
- Release validation recognizes nested skill directories (#830; closes #829).

### Verification

- Exact combined candidate: `OCTOPUS_NON_INTERACTIVE=1 make ci-local` — 16 smoke suites, 248 unit suites, and 7 integration suites passed.
- `make sync`, `make sync-check`, `./scripts/validate-release.sh 9.61.2`, `bash tests/unit/test-handoff.sh`, `git diff --check`, and executable-mode validation passed.
- Underlying current-head PR matrices passed on Ubuntu, macOS, the symlinked install path, and integration tests with CodeRabbit approval and zero unresolved threads.

After squash merge, the tag will be created on the exact main merge commit, followed by the GitHub Release, shared marketplace sync, and real Claude/Codex upgrade verification from v9.61.1.

Closes #836.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved model pin and allowlist handling across supported providers.
  * Added safer provider availability checks and recovery after smoke-test failures.
  * Improved lifecycle cleanup and non-interactive approval handling.

* **Documentation**
  * Updated release information, feature descriptions, and upgrade instructions.
  * Added guidance for marketplace auto-updates and plugin reloading.

* **Tests**
  * Expanded validation for skills and provider workflows.
  * Added coverage to ensure unit-test workflows have sufficient execution time.

* **Release**
  * Published version 9.61.2 with synchronized version information across supported integrations.

Closes #846.

### CODE REVIEW

**Comment by nyldn** [comment]
Re: CodeRabbit's "Linked Issues check" warning above — this is a release/version-bump PR, not a code-fix PR. The `validate_model_allowed()` case arms for #817 were already implemented and merged in #819, which is part of this branch's base (this PR only bumps version numbers, regenerates derived docs, and adds the CHANGELOG entry on top of that already-merged fix). No code change is missing here.

(Also: closed #821, a duplicate v9.61.2 release PR from an earlier session that predated #819's merge — this PR supersedes it.)


---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by nyldn** [comment]
Checked in from today's later triage run: this release PR is still green (Smoke/Unit/Integration/Portability all passing) and no new bug-fix commits have landed on `main` since it was opened. Leaving it for a human to merge rather than merging it myself — no action needed from this run.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by nyldn** [comment]
Heads up: this release PR is now stale relative to `main`. It's based on `3d3f972f`, but `main` has moved on with 4 more merged fixes since:

- #820 — restore Codex marketplace update compatibility
- #824 — make skill-template test hermetic
- #826 — council approval gates ignore non-interactive session signals
- #828 — lifecycle hook timeout cleanup self-contained (no more `pkill` dependency)

None of those are reflected in this PR's version bump or CHANGELOG entry. Rather than open a second, conflicting release PR for the same `v9.61.2` slot, flagging it here — a human should either:
1. Rebase this PR onto current `main` and fold the additional fixes into the CHANGELOG entry before merging, or
2. Merge this as-is for a quick `v9.61.2`, and let the next fixes go out in a follow-up `v9.61.3`.

Not merging or rebasing this myself per the daily-triage routine's release protocol (releases are left for a human).

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by nyldn** [comment]
Checked in from today's later triage run (after #833 merged): this release PR is now further behind `main`. In addition to the 4 fixes flagged in the previous comment (#820, #824, #826, #828), 2 more have landed since:

- #830 — release validator recognizes nested skill directories
- #833 — provider-timeout regression test no longer flakes on wall-clock jumps (this run's own fix)

That's 6 merged fixes since this PR's base (`3d3f972f`) not reflected in its version bump or CHANGELOG entry.

Same as the prior flag, not rebasing or merging this myself — releases are left for a human per the daily-triage routine's protocol. A human should either:
1. Rebase this PR onto current `main` and fold the additional fixes into the CHANGELOG entry before merging, or
2. Merge this as-is for a quick `v9.61.2` covering just #816/#819, and let the rest ship in a follow-up `v9.61.3`.

For reference, if going with option 1, here's a CHANGELOG entry draft covering all 8 fixes since v9.61.1 (#818, #830, #828, #826, #824, #819, #816, #833) that a human or the next session can drop in:

```markdown
### Fixed

- **Existing Codex marketplace installs can update again.** The Codex adapter
  preserves its original `claude-octopus` marketplace identity instead of
  borrowing Claude Code's separate `octo` selector. Shared-marketplace release
  sync now validates both host manifests, preventing stale Codex bundles from
  being stranded behind an identifier mismatch. (#818)
- **Release validator recognizes nested skill directories.**
  `scripts/validate-release.sh`'s skill-registration and frontmatter checks
  now see the supported one-level-nested `skills/octopus-starter-pack/*/SKILL.md`
  layout instead of reporting all four of its skills as missing. (#830)
- **Lifecycle hook timeout fallback no longer depends on missing `pkill`.**
  The built-in fallback (used when neither `run_with_timeout` nor a `timeout`
  binary is available) now signals the hook's whole process group via Bash
  job control, so forked children are reaped without an external `pkill`
  dependency. (#828)
- **Council approval gates fail closed under non-interactive PTY sessions.**
  `council_prompt_gate_approval` now checks the shared non-interactive
  detector (`CI`, `OCTOPUS_NON_INTERACTIVE`, `CLAUDE_CODE_REMOTE`, etc.)
  before its TTY check, so a PTY-backed automation session can no longer
  have stdin misread as an approval answer. (#826)
- **Skill-template test no longer mutates the tracked checkout.**
  `test-skill-templates.sh` now regenerates against a throwaway fixture
  instead of overwriting hand-authored `flow-*` skill files with template
  output. (#824)
- **`OCTOPUS_OLLAMA_ALLOWED_MODELS`, `OCTOPUS_COPILOT_ALLOWED_MODELS`, and
  `OCTOPUS_VIBE_ALLOWED_MODELS` are enforced.** `validate_model_allowed()`
  previously fell through to "allow" for these three providers; they now get
  the same allowlist enforcement as every other seat. (#819)
- **`OCTOPUS_OLLAMA_MODEL`, `OCTOPUS_COPILOT_MODEL`, and `OCTOPUS_VIBE_MODEL`
  pins take effect.** `get_agent_model()` had no case arm for these three
  agent types, so the env-override lookup built an unmatchable variable name
  and always fell back to the hardcoded default model. (#816)
- **Provider-timeout regression test no longer flakes on wall-clock jumps.**
  `test-provider-auth-validity.sh`'s SIGKILL-escalation test now measures
  elapsed time with a monotonic clock and asserts the SIGKILL exit code plus
  proof the process actually died, instead of a wall-clock duration that
  could read hundreds of seconds after a host clock adjustment. (#833)
```

---
_Generated by [Claude Code](https://claude.ai/code)_

File: AI_AGENT_HANDOFF.md [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-generator-safety, flow-define, and #804 skill-tree work. Issues #799-#801 and
-streamlining Parts 4b/4c/4d remain unstarted.
+Last updated: 2026-08-10
+Status: v9.61.2 is the active release, with permanent hook-timeout cleanup,
+careful-mode false-positive fixes, Codex marketplace update recovery, and
+provider availability/health hardening. The Gemini macOS keychain modal remains
+tracked as future issue #838; cancellation cleanup remains #841.
 Branch: `main`
-Release: [v9.61.1](https://github.com/nyldn/claude-octopus/releases/tag/v9.61.1)
+Release: [v9.61.2](https://github.com/nyldn/claude-octopus/releases/tag/v9.61.2)
```
[the remark this answers is not in the document]
Comment (nyldn) [reply AI_AGENT_HANDOFF.md]: Fixed in 4e2c26d7. The handoff now identifies v9.61.2 as the release candidate in PR #823, links the candidate rather than the unpublished Release, and makes the exact-main CI/tag/Release/marketplace publication sequence explicitly pending.

File: AI_AGENT_HANDOFF.md [hunk 2]
```diff
@@ -64,7 +65,89 @@ v49 with four pending migrations to v53. Repository rules reserve migration for
...
+- Careful-hook safety suite: 26/26 plus direct real-hook probes for quiet and
+  destructive cases.
+- Provider predicates: 13/13; Vibe 5/5; feature detection 20/20;
+  OpenAI-compatible agent 19/19.
+- Smoke dispatch exclusion: 9/9; quota watcher 2/2; embrace fail-fast 6/6;
+  Gemini provider 52/52.
+- Bash syntax, relevant ShellCheck gates, diff checks, and executable-mode
+  checks were clean. Test-created stale worktree records were pruned.
+
+## Delivered in v9.61.1 Cycle
```
[the remark this answers is not in the document]
Comment (nyldn) [reply AI_AGENT_HANDOFF.md]: Addressed in 4e2c26d7 by making every handoff status/heading explicitly candidate-scoped and preserving publication as the next action. CHANGELOG.md remains the prepared v9.61.2 release entry required by RELEASING.md; merge, exact-main CI, tag, and GitHub Release cannot occur before this release PR is approved and merged. I will complete that sequence after the current-head gates pass, then publish a follow-up handoff state if exact post-publication evidence needs a new commit.

File: AI_AGENT_HANDOFF.md [hunk 3]
```diff
@@ -64,7 +65,89 @@ v49 with four pending migrations to v53. Repository rules reserve migration for
...
+  the Node-based Gemini CLI. The issue records the screenshot text, suspected
+  detection/config paths, and non-prompting acceptance criteria. It is a future
+  task, not part of v9.61.2.
+
+Verification completed before the release candidate:
+
+- PR #833 final local `make ci-local`: 16 smoke, 247 unit, and 7 integration
+  suites passed; Linux, macOS, symlink, and integration checks also passed
+  remotely.
+- Lifecycle focused suites: 10/10 and provider auth timeout 18/18.
```
[the remark this answers is not in the document]
Comment (nyldn) [reply AI_AGENT_HANDOFF.md]: Fixed in 4e2c26d7: the verification bullet now says Lifecycle-focused suites.

File: tests/unit/test-suite-reachability.sh [hunk 4]
```diff
@@ -192,6 +192,18 @@ else
     test_fail "found only ${n_targets} 'make test-*' invocations in the workflow — the grep or the workflow changed, so the assertion above would be vacuous"
 fi
 
+test_case "the unit matrix timeout has headroom for slow macOS runners"
+unit_timeout_minutes="$(awk '
+    /^  unit:/ { in_unit = 1; next }
+    in_unit && /^  [[:alnum:]_-]+:/ { exit }
+    in_unit && /timeout-minutes:/ { print $2; exit }
+' "$WORKFLOW")"
```
[the remark this answers is not in the document]
Comment (nyldn) [reply tests/unit/test-suite-reachability.sh:200]: Fixed in 163d636b. The parser now requires the exact four-space job-level timeout key. Focused suite: 10/10; a synthetic unit job containing only a nested timeout produced an empty result (`nested_timeout_rejected=yes`).
